editor/plugins/tiles/tile_set_atlas_source_editor.cpp

```diff
@@ -1911,6 +1911,9 @@ void TileSetAtlasSourceEditor::_tile_alternatives_control_gui_input(const Ref<In
 			}
 		} else if (mb->get_button_index() == MouseButton::RIGHT) {
 			if (mb->is_pressed()) {
+				drag_type = DRAG_TYPE_MAY_POPUP_MENU;
+				drag_start_mouse_pos = alternative_tiles_control->get_local_mouse_position();
+			} else {
 				// Right click pressed
 				Vector3 tile = tile_atlas_view->get_alternative_tile_at_pos(mouse_local_pos);
```

Comment:
The popup of the menu should be conditioned by `if (drag_type == DRAG_TYPE_MAY_POPUP_MENU)`. The point is to only to popup the menu if you did not drag too far.